waspc/src/Wasp/Generator/WebAppGenerator.hs

```diff
@@ -135,10 +135,10 @@ npmDepsForWasp spec =
         AS.Dependency.fromList
           [ -- TODO: Allow users to choose whether they want to use TypeScript
             -- in their projects and install these dependencies accordingly.
-            ("vite", "^4.1.0"),
-            ("typescript", "^4.9.3"),
-            ("@types/react", "^17.0.53"),
-            ("@types/react-dom", "^17.0.19"),
+            ("vite", "^4.3.9"),
+            ("typescript", "^4.9.3"), -- Once upgraded to Typescript 5, consider using moduleResolution: 'bundler' in tsconfig.node.json, since that is what Vite now by default uses (we have it set to 'Node' right now because 'bundler' requires TS >= 5). Also, upgrade @tsconfig/vite-react to 2.0.0 (couldn't do it now because it also uses 'bundler').
```

Comment:
Maybe we merge @sodic's PR first and then we can do this as well in this PR?

Reply:
I merged the TS version update and upgraded `vite-react` to 2.0.0. We inherit from `vite-react`'s config, and version 2.0.0 sets `module: bundler`, so we're good.